app/presenters/tree_builder_roles_by_server.rb

```diff
@@ -0,0 +1,56 @@
+class TreeBuilderRolesByServer < TreeBuilder
+  has_kids_for MiqServer, [:x_get_tree_miq_server_kids]
+
+  def initialize(name, type, sandbox, build = true, parent = nil)
+    @root = parent
+    super(name, type, sandbox, build)
+  end
+
+  private
+
+  def tree_init_options(_tree_name)
+    {
+      :add_root => false,
+      :expand   => true,
+      :lazy     => false,
+      :open_all => true
+    }
+  end
+
+  def set_locals_for_render
+    locals = super
+    locals.merge!(:autoload  => false,
+                  :click_url => "/ops/diagnostics_tree_select/",
+                  :onclick   => "miqOnClickServerRoles")
+  end
+
+  def root_options
+    []
+  end
+
+  def x_get_tree_roots(count_only = false, _options)
+    count_only_or_objects(count_only, x_get_tree_miq_servers)
+  end
+
+  def x_get_tree_miq_servers
+    objects = []
+    @root.miq_servers.sort_by { |s| s.name.to_s }.each do |s|
```

Comment:
`each_with_object` instead of `each` will allow you to remove the 1st and last line of this methos.